Add SCC Registration page (#404)
rancher/rancher-product-docs: versions/latest/modules/en/pages/rancher-admin/scc-registration.adoc

= SUSE Customer Center Registration
:revdate: 2025-08-19
:page-revdate: {revdate}

> Since Rancher v2.12.1, you should now register your SUSE Rancher Manager Prime Suite with https://scc.suse.com/home[SUSE Customer Center (SCC)] from the Rancher UI. This will enable you to see your registered systems in SCC under your subscription.

Author: @btat Based on feedback from Patrick, I changed the wording of the intro to `SUSE Rancher Prime customers can now register Rancher Manager with [SUSE Customer Center` d7eca854c255107f4654a6f3e424831f4bd1773c

== Prequsities

> Registering your Rancher Manager instance with SUSE Customer Center requires a SUSE account. If you don't have a SUSE account yet, visit the https://scc.suse.com/[SCC home page] to create one.

Reviewer: ```suggestion
Registering your Rancher Manager instance with SCC requires a SUSE account. If you don't have a SUSE account yet, visit the https://scc.suse.com/[SCC home page] to create one.
```

The acronym should be used since you already defined it and use it for other cases.

> You must also have access to your organization. To join your organization, you may either:

Reviewer: This skips the case where someone is the first user/admin and there's no existing organization.

Author: @mallardduck @l-mb should this scenario be added?

> * Ask your organization's administrator to add you to the account. See our https://scc.suse.com/docs/userguide#UG-Requesting-Access-to-an-Organizations-Account[documentation] for details.

Reviewer: ```suggestion
* Ask your organization's administrator to add you to the account. See the https://scc.suse.com/docs/userguide#UG-Requesting-Access-to-an-Organizations-Account[SCC documentation] for details.
```

> * If you know one of your organization's registration codes, enter it when prompted.

Reviewer: How does someone get or find a registration code?

Author: @btat This is referring to a registration code for access to an organization https://scc.suse.com/docs/userguide#UG-Requesting-Access-to-an-Organizations-Account. Do you think I should remove `If you know one of your organization's registration codes, enter it when prompted.` to prevent confusion? 

```suggestion
* You must have access to your organization. Ask your organization's administrator to add you to the account. See the https://scc.suse.com/docs/userguide#UG-Requesting-Access-to-an-Organizations-Account[SCC documentation] for details.
```

Reviewer: @LucasSaintarbor I think if the [FAQ scenario](https://github.com/rancher/rancher-product-docs/pull/404#discussion_r2294917889) is covered then the above suggestion works.

> 

Reviewer: Having a registration code should be listed as one of the prereqs since users need it to register in Rancher.

Author: @btat I moved this prereq to online reg only via online reg only via 32fa013dd4f938e56aaf9af926ebdd374adf66d1

== Register with SUSE Customer Center

Check the Notification Center (bell icon) in the upper-right corner of your Rancher dashboard for a "Rancher Prime Registration" notification and click *Register Now*. This redirects you to the *Registration* page under *Global Settings*. 

You have the option to register your Rancher Manager instance with SCC <<online,Online Registration>> or <<offline,Offline Registration>> from the Rancher UI.

=== Online Registration

You can register online through the Rancher UI or with Helm.

==== Rancher UI

. Get the Registration code for your Rancher Manager subscription on SCC. Log in to SCC, navigate to **My Organizations** and click on the organization with the Rancher Manager subscription. 

> . Find your Registration code under *Organization* > *Subscriptions* > *Subscription Information*. 

Reviewer: These steps to get a registration code should be part of the prerequisites since it's needed regardless of the method used to register.

There's also some inconsistency right now because these steps are only part of the UI section, but not part of the Helm section.

==== Helm

In `values.yaml`, enable registration and add your registration code.

[,yaml]
----
registration:
  enabled: true
  regCode: ""
----

=== Offline Registration

If you are using Rancher in an air-gapped environment, you can register offline through the Rancher UI or with Helm.

==== Rancher UI

. Click *Download registration request* to download the registration request file.
. Click *Visit SCC* to use the SCC offline registration tool. 
.. Upload the registration request file.
.. Select your *Organization* and *Subscription*.
.. Click *Download Offline Certificate* and save the registration certificate file.
. Return to your Rancher Manager UI, click *Upload certificate and register* to upload the registration certificate. If successful, the *Status* will be *Active*, and the *Product Name*, *Expiration date* and *Registration mode* will be displayed.

==== Helm

. In `values.yaml`, enable registration and set the `mode` to `offline`.
+
[,yaml]
----
registration:
  enabled: true

>   regCode: ""

Reviewer 2: For offline mode, they comment out or remove `regCode`.

Author: @mallardduck Fixed this via 65bcc4eacfd198655e2fe52e8de0dcffbef1a193

Reviewer 2: lgtm ty

  mode: offline
----

> . Using `kubectl`, retrieve the registration request file and save it locally.

Reviewer 2: For instance, this command uses Bash to wait for up to 5 minutes until the secret exists. Once it exists it will exist and say "Secret Created!" - then the next steps could proceed.

```
for i in {1..100}; do kubectl wait --for=create secret/scc-registration -n cattle-scc-system --timeout=3s &> /dev/null && echo "Secret created!" && break || { echo "Not ready yet..."; sleep 3; }; done
```

= SUSE Customer Center Registration
:revdate: 2025-08-19
:page-revdate: {revdate}

> Since Rancher v2.12.1, you should now register your SUSE Rancher Manager Prime Suite with https://scc.suse.com/home[SUSE Customer Center (SCC)] from the Rancher UI or through Helm. This will enable you to see your registered systems in SCC under your subscription. 

Reviewer: `you should` - is registration a recommendation or mandatory?

Reviewer: @LucasSaintarbor I'm going to add an approval since the phrasing change by https://github.com/rancher/rancher-product-docs/pull/404/commits/d7eca854c255107f4654a6f3e424831f4bd1773c skirts around this, but we should still find an answer because it's likely something that will be asked.

> Since Rancher v2.12.1, you should now register your SUSE Rancher Manager Prime Suite with https://scc.suse.com/home[SUSE Customer Center (SCC)] from the Rancher UI or through Helm. This will enable you to see your registered systems in SCC under your subscription. 

Reviewer: ```suggestion
Since Rancher v2.12.1, you should now register your SUSE Rancher Manager Prime Suite with https://scc.suse.com/home[SUSE Customer Center (SCC)] from the Rancher UI or through Helm. This enables you to see your registered systems in SCC under your subscription. 
```

== Prequsities

> * Registering your Rancher Manager instance with SCC requires a SUSE account. If you don't have a SUSE account yet, visit the https://scc.suse.com/[SCC home page] to create one.

Reviewer: ```suggestion
* Registering your Rancher Manager instance with SCC requires a SUSE account. If you don't have a SUSE account yet, [create one](https://www.suse.com/account/create/).
```

The register button on the SCC homepage has a direct link.

* You must have access to your organization. Ask your organization's administrator to add you to the account. See the https://scc.suse.com/docs/userguide#UG-Requesting-Access-to-an-Organizations-Account[SCC documentation] for details. If you know your organization's registration code, enter it when prompted.

* You must have the registration code for your Rancher Manager subscription on SCC. To get the registration code:
. Log in to SCC.
. Navigate to **My Organizations** and click on the organization with your Rancher Manager subscription.
. Find your Registration code under *Organization* > *Subscriptions* > *Subscription Information*.

== Register with SUSE Customer Center

> From the Rancher UI, check the Notification Center (bell icon) in the upper-right corner for a "Rancher Prime Registration" notification and click *Register Now*. This redirects you to the *Registration* page under *Global Settings*. You can also directly go to *Global Settings* > *Registration*.

Reviewer: We should list the actual navigation path as the default/first choice. The last sentence says it's the direct path, but we're listing the indirect option first.

Also, if the "Rancher Prime Registration" notification behaves like other notifications (is it a special notification?) then it can disappear after someone marks it as read.

You have the option to register your Rancher Manager instance with SCC <<online,Online Registration>> or <<offline,Offline Registration>>.

=== Online Registration

You can register online through the Rancher UI or with Helm.

==== Rancher UI

> . From the Rancher UI, copy the registration code into the *Registration code* field and click *Register*. If successful, the *Status* will be *Active*, and the *Product Name*, *Expiration date* and *Registration mode* will be displayed.

Reviewer: ```suggestion
. From the Rancher UI, copy the registration code into the *Registration code* field and click *Register*. If successful, the *Status* becomes *Active*, and the *Product Name*, *Expiration date* and *Registration mode* are displayed.
```

==== Helm

> Once you have your registration code, you can register your Rancher Prime system through Helm by passing the argument `--set registration.enabled=true --set registration.regCode=<REGISTRATION-CODE-FROM_SCC>` in the Helm install command `helm install rancher rancher-prime/rancher`.

Reviewer: ```suggestion
After you have your registration code, you can register your Rancher Prime system through Helm by passing the argument `--set registration.enabled=true --set registration.regCode=<REGISTRATION-CODE-FROM_SCC>` in the Helm install command `helm install rancher rancher-prime/rancher`.
```

You can also update `values.yaml`,  enabling registration and adding your registration code.

[,yaml]
----
registration:
  enabled: true
  regCode: ""
----

=== Offline Registration

If you are using Rancher in an air-gapped environment, you can register offline through the Rancher UI or with Helm.

==== Rancher UI

. From the Rancher UI, click *Download registration request* to download the registration request file.
. Click *Visit SCC* to use the SCC offline registration tool. 
.. Upload the registration request file.
.. Select your *Organization* and *Subscription*.
.. Click *Download Offline Certificate* and save the registration certificate file.

> . Return to the Rancher UI, click *Upload certificate and register* to upload the registration certificate. If successful, the *Status* will be *Active*, and the *Product Name*, *Expiration date* and *Registration mode* will be displayed.

Reviewer: ```suggestion
. Return to the Rancher UI, click *Upload certificate and register* to upload the registration certificate. If successful, the *Status* becomes *Active*, and the *Product Name*, *Expiration date* and *Registration mode* are displayed.
```

= SUSE Customer Center Registration
:revdate: 2025-08-19
:page-revdate: {revdate}

> Since Rancher v2.12.1, you should now register your SUSE Rancher Manager Prime Suite with https://scc.suse.com/home[SUSE Customer Center (SCC)] from the Rancher UI. This will enable you to see your registered systems in SCC under your subscription.

Author: @IsaSih I clarified Helm is an option and updated the online Helm steps via c7d0e1f51129202bfc2ef1d70be1cc0d3c50e3f2. Does this look right?

> In Rancher v2.12.1, you can now register your SUSE Rancher Manager Prime Suite with https://scc.suse.com/home[SUSE Customer Center (SCC)] from the Rancher UI. This will enable you to access the add-ons included in subscriptions purchased by your organization, open support cases and download the latest software updates and patches. 

Author: @l-mb Thanks for clarifying. Fixed via 5867ea717fe970f2bbcf6f91e481eac2ac951ebe

> In Rancher v2.12.1, you can now register your SUSE Rancher Manager Prime Suite with https://scc.suse.com/home[SUSE Customer Center (SCC)] from the Rancher UI. This will enable you to access the add-ons included in subscriptions purchased by your organization, open support cases and download the latest software updates and patches. 

Author: @l-mb based on Billy's [comment](https://github.com/rancher/rancher-product-docs/pull/404#discussion_r2294922743), is registering with SCC mandatory?  And from other feedback, is it okay to reference "access to add-ons"? 

Since Rancher v2.12.1, SUSE Rancher Prime customers can now register Rancher Manager with [SUSE Customer Center (SCC)](https://scc.suse.com/home). This registration provides access to add-ons, software updates and patches included in your organization's subscriptions. To learn how to register your Rancher Manager instance with SCC, refer to [documentation placeholder](documentation-url).

== Prequsities

Registering your Rancher Manager instance with SUSE Customer Center requires a SUSE account. If you don't have a SUSE account yet, visit https://scc.suse.com/[SCC home page] to create one.

You must also have access your organization. To join your organization, you may either:

* Ask your organization's administrator to add you to the account. See our https://scc.suse.com/docs/userguide#UG-Requesting-Access-to-an-Organizations-Account[documentation] for details.

> * If you know one of your organization's registration codes, enter it when promted.

Reviewer 3: "prompted"

== Register with SUSE Customer Center

> Check the Notification Center (bell icon) in the upper-right corner of your Rancher dashboard for a "Rancher Prime Registration" notification and click *Register Now*. You will be redirected to the *Registration* page under *Global Settings*. 

Reviewer 4: Also, it's useful to refer to the path in the UI that leads to the page, without needing to solely rely on the notification center. The path is `Global Settings > Registration`

> You have the option to register your Rancher Manager instance with SCC <<online,Online Registration>> or <<offline,Offline Registration>> from the Rancher UI.

Author: @IsaSih done. Using Rancher UI through out the page.

=== Online Registration

. Get the Registration code for your Rancher Manager subscription on SCC. Log in to SCC, go to **My Organizations** and click on the organization you want to access. 
. Find your Registration code under *Organization* > *Subscriptions* > *Subscription Information*. 
. Copy the registration code into the *Registration code* field and click *Register*. If successful, the *Status* will be *Active*, and the *Product Name*, *Expiration date* and *Registration mode* will be displayed.

=== Offline Registration

If you are using Rancher in an air-gapped environment, you can register offline. 

> . Click *Download registration request* to download the registration request file.

Reviewer 4: good to mention that this is done from the rancher UI

Author: @IsaSih Done.
